app/services/vine_voucher_validator_service.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+class VineVoucherValidatorService
+  attr_reader :voucher_code, :errors
+
+  def initialize(voucher_code:, enterprise:)
+    @voucher_code = voucher_code
+    @enterprise = enterprise
+    @errors = {}
+  end
+
+  def validate
+    if vine_settings.nil?
+      errors[:vine_settings] = I18n.t("vine_voucher_validator_service.errors.vine_settings")
+      return nil
+    end
+
+    response = call_vine_api
+
+    if !response.success?
+      handle_errors(response)
+      return nil
+    end
+
+    save_voucher(response)
+  rescue Faraday::Error => e
+    Rails.logger.error e.inspect
+    Bugsnag.notify(e)
+
+    # TODO do we need a more specific error ?
```

Comment:
We want to know if this is a network error, and therefore it may be intermittent and we can try again. It seems unlikely that Faraday would raise non-network errors, so this is probably specific enough. 👍 

But I suggest the `rescue` should be closer to where the request is called, in `call_vine_api`

Reply:
If I do that, I'll need to handle the case where `call_vine_api` doesn't return a `response` object, it's not complicated but it's more code. 
What's your issue with the `rescue` being  there ? Given the method name 'call_vine_api' it should be pretty clear where the exception is coming from.

Reply:
I think in general it's easier to follow and leaves less room for unexpected outcomes. For example what if we make multiple faraday requests in the future, and the `rescue` code is assuming one but not the other.
But as you say it should be pretty clear, and it's not worth making the code more complicated.